harness/determined/exec/prep_container.py

```diff
@@ -308,6 +308,12 @@ def set_proxy_address(sess: api.Session, allocation_id: str) -> None:
 
 
 def set_proxy_address_kubernetes(sess: api.Session, allocation_id: str) -> None:
+    # When using a gateway, we know the static IP in advance so we don't want to tell the
+    # master the pod IP. Handling this on the master side gets a little tricky with restore.
+    # For example we send this proxy address before master sends the allocation its address.
+    if os.environ.get("DET_PROXY_THROUGH_GATEWAY") == "true":
```

Comment:
`if bool(os.environ.get("DET_PROXY_THROUGH_GATEWAY"):` to accept truth-y values that aren't exactly "true"